### PR TITLE
 Fix arcadia spark run configuration related bugs

### PR DESCRIPTION
**Pull request by t-rufang**

### CODE REVIEW

**Comment by lcadzy** [comment]
Thank you! Rui

**Comment by lcadzy** [comment]
> class CosmosServerlessSparkAccountsCombo: SparkClusterListRefreshableCombo() {

please check whether these related refresh combo can work

---
Refers to: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkClusterListRefreshableCombo.kt:223 in c0d1b2d. [](commit_id = c0d1b2dbb8768ea8ed5b5577a8b9bc33c796183f, deletion_comment = False)

**Comment by t-rufang** [comment]
> class CosmosServerlessSparkAccountsCombo: SparkClusterListRefreshableCombo() {

Already test locally. Different spark application work fine.

---
In reply to: [521868511](https://github.com/microsoft/azure-tools-for-java/pull/3412#issuecomment-521868511) [](ancestors = 521868511)

---
Refers to: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkClusterListRefreshableCombo.kt:223 in c0d1b2d. [](commit_id = c0d1b2dbb8768ea8ed5b5577a8b9bc33c796183f, deletion_comment = False)

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitModel.java [hunk 1]
```diff
@@ -53,6 +53,8 @@
     @NotNull
     private SparkSubmissionParameter submissionParameter; // The parameters to packing submission related various
 
+    private String clusterTitle;
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitModel.java]: `@Nullable`?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkClusterListRefreshableCombo.kt [hunk 2]
```diff
@@ -123,7 +126,7 @@ open class SparkClusterListRefreshableCombo: ILogger, Disposable {
             BehaviorSubject.create(ImmutableComboBoxModel(initClusters))
         }
 
-        val toSelectClusterByIdBehavior: BehaviorSubject<Any> = disposableSubjectOf { BehaviorSubject.create() }
+        val toSelectClusterByIdBehavior: BehaviorSubject<SelectedClusterProperty> = disposableSubjectOf { BehaviorSubject.create() }
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkClusterListRefreshableCombo.kt]: Why not `IClusterDetail`?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkClusterListRefreshableCombo.kt [hunk 3]
```diff
@@ -64,6 +64,8 @@ open class SparkClusterListRefreshableCombo: ILogger, Disposable {
 
     open val comboBoxName: String = "clusterListComboBox"
 
+    data class SelectedClusterProperty(val name: String, val title: String)
```
Comment (t-rufang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkClusterListRefreshableCombo.kt]: SelectClusterEvent

**Pull request merged**